src/sentence.rs

```diff
@@ -0,0 +1,133 @@
+use std::cell::{RefCell, RefMut};
+use std::rc::Rc;
+
+use conllx::graph::{Node, Sentence};
+use conllx::token::{Token, TokenBuilder};
+use pyo3::class::basic::PyObjectProtocol;
+use pyo3::class::sequence::PySequenceProtocol;
+use pyo3::exceptions;
+use pyo3::prelude::*;
+
+/// Sentence that can be annotated.
+#[pyclass(name=Sentence)]
+pub struct PySentence {
+    inner: Rc<RefCell<Sentence>>,
+}
+
+#[pymethods]
+impl PySentence {
+    /// Construct a new sentence from forms and (optionally) POS tags.
+    ///
+    /// The constructor will throw a `ValueError` if POS tags are
+    /// provided, but the number or tags is not equal to the number of
+    /// tokens.
+    #[new]
+    fn __new__(obj: &PyRawObject, forms: Vec<&str>, pos_tags: Option<Vec<&str>>) -> PyResult<()> {
```

Comment:
~~~python
s = Sentence("xyz")
for t in s:
    print(t.form)
~~~

will produce:

~~~
None
x
y
z
~~~

Is there any type of check we could do here to prevent that?

Reply:
Pretty nasty

```
>>> type("abc"[0])
type("abc"[0])
```

But this should be simple to check by verifying that the outer type is not a string.

Reply:
I am not sure whether we want this though, since this might be intended behavior in Python?

Reply:
Ok, I have this working locally, but there are several solutions:

* Strict: only accept a list of `str`. This rejects a lot of problematic cases, but also some borderline cases (tuple of `str`).
* Lenient: accept any sequence that is not `str`.

Reply:
I have now implemented a version that verifies that the type is a Python list -- the value type should be convertable to `String`.